### PR TITLE
STORM-2863: Reset NormalizedResources for tests

### PR DESCRIPTION
**Pull request by revans2**

### CODE REVIEW

**Comment by srdo** [comment]
@revans2 Could you take a look at https://github.com/apache/storm/pull/2463? I think it will resolve this issue too.

**Comment by revans2** [comment]
@srdo yes #2463 would resolve the issue too, but it loses most of the reasons why I put in NormalizedResources in the first place.  Lets move the discussion over there for now.

File: storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java [hunk 1]
```diff
@@ -65,11 +67,33 @@
...
+            return result;
+        }
+    };
+
     /**
      * test if the scheduling logic for the DefaultResourceAwareStrategy is correct
      */
     @Test
     public void testDefaultResourceAwareStrategySharedMemory() {
+        NormalizedResources.resetResourceNames();
```
Comment (srdo) [inline storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java]: Isn't it better to put this in `@Before` or `@After` (or ideally a `TestRule`)? I think it's likely we'll forget to copy this to new tests.

Comment (revans2) [reply storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java]: Good point I will update it

**Comment by srdo** [comment]
Does https://github.com/revans2/incubator-storm/blob/6aaf8ee7cb042960831547a3ef07ce68a1105970/storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestGenericResourceAwareStrategy.java also affect the static fields?

**Comment by revans2** [comment]
@srdo I moved it over to being a Rule annotation

**Comment by srdo** [comment]
Thanks, +1 assuming TestGenericResourceAwareStrategy doesn't touch the static fields.

**Pull request merged**